StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentElement+Internal.swift

```diff
@@ -127,14 +127,16 @@ extension EmbeddedPaymentElement: EmbeddedPaymentMethodsViewDelegate {
         // Only show forms that require user input
         guard formViewController.collectsUserInput else {
             self.formViewController = nil  // Clear out any previous form view controller to update self._paymentOption
-            return
+            return true
         }
 
         let bottomSheet = bottomSheetController(with: formViewController)
         delegate?.embeddedPaymentElementWillPresent(embeddedPaymentElement: self)
         presentingViewController.presentAsBottomSheet(bottomSheet, appearance: configuration.appearance)
         self.formViewController = formViewController
+        return formViewController.selectedPaymentOption != nil // Don't show row as selected when presenting a form unless we have a valid payment option
```

Comment:
Nit, this might improve readability
```
        let formHasValidPaymentOption = formViewController.selectedPaymentOption != nil
        return formHasValidPaymentOption // Show row selected only if payment option is valid
```

Reply:
I may have found a bug... It's possible to fill out a payment method, then tap outside the sheet, which results in no payment method being selected.  Will DM you with a video

Reply:
Good catch on this bug! It actually has some unwanted behavior on master too. The fix was quite simple since we have access to the previous payment option on the form VC.